src/wm_info_provider/river.rs

```diff
@@ -184,11 +194,38 @@ fn output_status_cb(ctx: EventCtx<State, ZriverOutputStatusV1>) {
 }
 
 fn seat_status_cb(ctx: EventCtx<State, ZriverSeatStatusV1>) {
-    if let zriver_seat_status_v1::Event::Mode(mode) = ctx.event {
-        let river = ctx.state.shared_state.get_river().unwrap();
-        let mode = mode.to_string_lossy().into_owned();
-        river.mode = (mode != "normal").then_some(mode);
-        ctx.state.mode_name_updated(ctx.conn, None);
+    match ctx.event {
+        zriver_seat_status_v1::Event::Mode(mode) => {
+            let river = ctx.state.shared_state.get_river().unwrap();
+            let mode = mode.to_string_lossy().into_owned();
+            river.mode = (mode != "normal").then_some(mode);
+            ctx.state.mode_name_updated(ctx.conn, None);
+        }
+        zriver_seat_status_v1::Event::FocusedOutput(output) => {
+            let river = ctx.state.shared_state.get_river().unwrap();
+            let Some(status) = river
+                .output_statuses
+                .iter_mut()
+                .find(|s| s.output == output)
+            else {
+                return;
+            };
+            status.is_focused = true;
+            ctx.state.tags_updated(ctx.conn, None);
```

Comment:
```suggestion
            ctx.state.tags_updated(ctx.conn, Some(status.output));
```

so that only the affected output is rendered. Also this will unnecessarily recompute tags button layout, so I would recommend adding something like this to `state.rs`:

```rust
pub fn output_focus_updated(&mut self, conn: &mut Connection<Self>, output: WlOutput) {
    self.for_each_bar(Some(output), |bar, ss| {
        bar.frame(conn, ss);
    });
}
```